vector_store.go

```diff
@@ -0,0 +1,349 @@
+func (c *Client) DeleteVectorStore(
+	ctx context.Context,
+	vectorStoreID string,
+) (response VectorStoreDeleteResponse, err error) {
+	urlSuffix := fmt.Sprintf("%s/%s", vectorStoresSuffix, vectorStoreID)
+	req, _ := c.newRequest(ctx, http.MethodDelete, c.fullURL(urlSuffix),
+		withBetaAssistantVersion(c.config.AssistantVersion))
+
+	err = c.sendRequest(req, &response)
+	return
+}
+
+// ListVectorStores Lists the currently available vector store.
+func (c *Client) ListVectorStores(
+	ctx context.Context,
+	limit *int,
+	order *string,
+	after *string,
+	before *string,
+) (response VectorStoresList, err error) {
+	urlValues := url.Values{}
+
+	if after != nil {
+		urlValues.Add("after", *after)
+	}
+	if order != nil {
+		urlValues.Add("order", *order)
+	}
+	if limit != nil {
+		urlValues.Add("limit", fmt.Sprintf("%d", *limit))
+	}
+	if before != nil {
+		urlValues.Add("before", *before)
+	}
+
+	encodedValues := ""
+	if len(urlValues) > 0 {
+		encodedValues = "?" + urlValues.Encode()
+	}
+
+	urlSuffix := fmt.Sprintf("%s%s", vectorStoresSuffix, encodedValues)
+	req, _ := c.newRequest(ctx, http.MethodGet, c.fullURL(urlSuffix),
+		withBetaAssistantVersion(c.config.AssistantVersion))
+
+	err = c.sendRequest(req, &response)
+	return
+}
+
+// CreateVectorStoreFile creates a new vector store file.
+func (c *Client) CreateVectorStoreFile(
+	ctx context.Context,
+	vectorStoreID string,
+	request VectorStoreFileRequest,
+) (response VectorStoreFile, err error) {
+	urlSuffix := fmt.Sprintf("%s/%s%s", vectorStoresSuffix, vectorStoreID, vectorStoresFilesSuffix)
+	req, _ := c.newRequest(ctx, http.MethodPost, c.fullURL(urlSuffix),
+		withBody(request),
+		withBetaAssistantVersion(c.config.AssistantVersion))
+
+	err = c.sendRequest(req, &response)
+	return
+}
+
+// RetrieveVectorStoreFile retrieves a vector store file.
+func (c *Client) RetrieveVectorStoreFile(
+	ctx context.Context,
+	vectorStoreID string,
+	fileID string,
+) (response VectorStoreFile, err error) {
+	urlSuffix := fmt.Sprintf("%s/%s%s/%s", vectorStoresSuffix, vectorStoreID, vectorStoresFilesSuffix, fileID)
+	req, _ := c.newRequest(ctx, http.MethodGet, c.fullURL(urlSuffix),
+		withBetaAssistantVersion(c.config.AssistantVersion))
+
+	err = c.sendRequest(req, &response)
+	return
+}
+
+// DeleteVectorStoreFile deletes an existing file.
+func (c *Client) DeleteVectorStoreFile(
+	ctx context.Context,
+	vectorStoreID string,
+	fileID string,
+) (err error) {
+	urlSuffix := fmt.Sprintf("%s/%s%s/%s", vectorStoresSuffix, vectorStoreID, vectorStoresFilesSuffix, fileID)
+	req, _ := c.newRequest(ctx, http.MethodDelete, c.fullURL(urlSuffix),
+		withBetaAssistantVersion(c.config.AssistantVersion))
+
+	err = c.sendRequest(req, nil)
+	return
+}
+
+// ListVectorStoreFiles Lists the currently available files for a vector store.
+func (c *Client) ListVectorStoreFiles(
+	ctx context.Context,
+	vectorStoreID string,
+	limit *int,
+	order *string,
+	after *string,
+	before *string,
```

Comment:
Could we please use the [Pagination](https://github.com/sashabaranov/go-openai/blob/211cb49fc22766f4174fef15301c4d39aef609d3/run.go#L210) struct here?